modules/compiler/utils/source/barrier_regions.cpp

```diff
@@ -406,26 +411,50 @@ void compiler::utils::Barrier::Run(llvm::ModuleAnalysisManager &mam) {
     node.id = kBarrier_FirstID;
     node.successor_ids.push_back(kBarrier_EndID);
     kernel_id_map_[kBarrier_FirstID] = &func_;
-  } else {
-    // if we found some barriers, need to split up our kernel across them!
-    {
-      ModulePassManager pm;
-      // It's convenient to create LCSSA PHI nodes to stop values defined
-      // within a loop being stored to the barrier unnecessarily on every
-      // iteration (if, for instance, the loop is entirely between two
-      // barriers, but the value is used outside of that barrier region).
-      pm.addPass(llvm::createModuleToFunctionPassAdaptor(LCSSAPass()));
-      pm.run(module_, mam);
-      mam.invalidate(module_, PreservedAnalyses::allInSet<CFGAnalyses>());
-    }
-    // do the splitting first in case a value is used on both sides of a
-    // barrier within the same basic block.
-    SplitBlockwithBarrier();
-    FindLiveVariables();
+    return;
+  }
+
+  // If we found some barriers, we need to split up our kernel across them!
```

Comment:
is the else no longer relevant here? the else seems to reflect finding barriers which this comment suggests it is assuming.